src/main/java/com/telran/libraryapp/entity/Visitor.java

```diff
@@ -20,18 +17,38 @@
 @Entity
 public class Visitor {
     @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
```

Comment:
Visitor primary key  according to the scheme should be "email" field, please remove Long id field and use email as primary key in the table